### ISSUE TITLE
One less thread available than expected in TaskPool Resource.

### ISSUE THREAD

**Issue by GrantMoyer** [bevyengine/bevy#416]
Even though ComputeTaskPool::thread_num() reports 4, apparently only 3 are used for work when accessing the ComputeTaskPool resource. Maybe one of the threads in the Resource is inadvertently blocked.

Affects master branch. Does not affect AsyncComputeTaskPool resource or IOTaskPool resource.

Code to reproduce:

```rust
use bevy::{
    prelude::*,
    tasks::{prelude::*, TaskPoolBuilder},
};
use std::{
    io::{stdout, Write},
    thread,
    time::{Duration, Instant},
};

fn task_system(pool: Res<ComputeTaskPool>) {
    let t0 = Instant::now();
    println!("{} threads", pool.thread_num());
    pool.scope(|s| {
        for i in 0..8 {
            s.spawn(async move {
                print!("{}", i);
                stdout().flush().unwrap();
                thread::sleep(Duration::from_secs(1));
            })
        }
    });
    let t1 = Instant::now();
    println!("\nTook {:.3}s", (t1 - t0).as_secs_f32());
}

fn main() {
    let pool = ComputeTaskPool(TaskPoolBuilder::new().num_threads(4).build());
    let t0 = Instant::now();
    println!("{} threads", pool.thread_num());
    pool.scope(|s| {
        for i in 0..8 {
            s.spawn(async move {
                print!("{}", i);
                stdout().flush().unwrap();
                thread::sleep(Duration::from_secs(1));
            })
        }
    });
    let t1 = Instant::now();
    println!("\nTook {:.3}s", (t1 - t0).as_secs_f32());

    App::build().add_startup_system(task_system.system()).run();
}
```

Output:

```
4 threads
17405326
Took 2.001s
4 threads
02631745
Took 3.001s
```

Expected output:

```
4 threads
17405326
Took 2.001s
4 threads
02631745
Took 2.001s
```

**Comment by GrantMoyer** [comment]
Comment from @aclysma on discord:

> that's expected right now, only 3 threads are available for your work because one of those threads is blocked by parallel_executor doing a blocking wait on a channel

So it looks like this is caused by #405.

**Comment by Cupnfish** [comment]
Now run your code on my computer, the output is this, I think this bug is solved.
```
4 threads
02475613
Took 2.009s
6 threads
01475623
Took 2.014s
```

**Comment by GrantMoyer** [comment]
Fixed by #437.
